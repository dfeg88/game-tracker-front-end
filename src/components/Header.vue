<template>
  <v-app-bar
      app
      color="primary"
      dark
      src="https://gamecrate.com/sites/default/files/field/image/PS5Header.jpg"
  >
    <div class="d-flex align-center">
      <router-link :to="{name: 'games'}">
        <v-img
            class="shrink mr-2"
            contain
            dark
            src="../assets/logo.png"
            transition="scale-transition"
            width="70"
        />
      </router-link>
    </div>

    <v-spacer></v-spacer>

    <v-btn plain exact :to="{name: 'games'}">
     My Games
    </v-btn>

    <v-btn plain :to="{name: 'create'}">
     Add Game
    </v-btn>

  </v-app-bar>
</template>

<script>
export default {
  name: "Header"
}
</script>

<style scoped>

</style>