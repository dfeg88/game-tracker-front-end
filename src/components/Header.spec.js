import Header from "@/components/Header";
import {createLocalVue, mount, shallowMount} from "@vue/test-utils";
import Vuetify from "vuetify";
import VueRouter from 'vue-router'

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()

let wrapper;
const vuetify = new Vuetify();

describe('Header', () => {
  beforeEach(() => {
    wrapper = shallowMount(Header, {
      localVue,
      router,
      vuetify
    });
  })

  it('renders expect HTML', () => {
    expect(wrapper.element).toMatchSnapshot();
  })
})