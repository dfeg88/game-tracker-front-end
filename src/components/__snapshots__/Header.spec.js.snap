// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders expect HTML 1`] = `
<v-app-bar-stub
  app="true"
  color="primary"
  dark="true"
  extensionheight="48"
  src="https://gamecrate.com/sites/default/files/field/image/PS5Header.jpg"
  tag="header"
  value="true"
>
  <div
    class="d-flex align-center"
  >
    <router-link-stub
      ariacurrentvalue="page"
      event="click"
      tag="a"
      to="[object Object]"
    >
      <v-img-stub
        class="shrink mr-2"
        contain="true"
        dark="true"
        options="[object Object]"
        position="center center"
        src="../assets/logo.png"
        transition="scale-transition"
        width="70"
      />
    </router-link-stub>
  </div>
   
  <v-spacer-stub />
   
  <v-btn-stub
    activeclass=""
    exact="true"
    plain="true"
    tag="button"
    to="[object Object]"
    type="button"
  >
    
   My Games
  
  </v-btn-stub>
   
  <v-btn-stub
    activeclass=""
    plain="true"
    tag="button"
    to="[object Object]"
    type="button"
  >
    
   Add Game
  
  </v-btn-stub>
</v-app-bar-stub>
`;
